### PR TITLE
 Fix bug that requests redistribute motion on no-redistributable column

### PR DESCRIPTION
**Pull request by khannaekta**
Prior to this commit, the function `PdrgpcrHashableSubset()` checked for
`FHashable()` instead of `FRedistributable()`. Updated the function as
well as the function name.
This PR also removes unused function `PdrgpcrHashablePrefix()`.

This PR resolves GPDB issue: https://github.com/greenplum-db/gpdb/issues/4529

### CODE REVIEW

**Comment by vraghavan78** [comment]
should CPhysicalUnionAll::PdshashedPassThru also think about redistributable instead of hashable?

**Comment by khannaekta** [comment]
@vraghavan78 We removed the check for FHashable() in `CPhysicalUnionAll::PdshashedPassThru()` and ctests all passed.

**Comment by vraghavan78** [comment]
You can use the similar example:

create table testhstore (h hstore);
explain (select h from testhstore) UNION ALL (select h from testhstore);

**Comment by khannaekta** [comment]
The required distribution spec for `UnionAll` will never be Hash distributed if the column is not redistributable.
So the condition in `CPhysicalSerialUnionAll.cpp`L100 (`if (0 == ulOptReq && CDistributionSpec::EdtHashed == pdsRequired->Edt())`) will not be satisfied, so the `CPhysicalUnionAll ::PdshashedPassThru()` function will not be called.

**Comment by vraghavan78** [comment]
lets chat about this in the tech review. I am confused and I am done for the day.

File: data/dxl/minidump/DQA-NonRedistributableCol.mdp [hunk 1]
```diff
@@ -0,0 +1,331 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Objective: A query with DQA on non-redistributable column should not request for Redistribute Motion 
+
+create table testhstore (h hstore);
```
Comment (bhuvnesh2703) [inline data/dxl/minidump/DQA-NonRedistributableCol.mdp]: how do we create the hstore type?

**Comment by bhuvnesh2703** [comment]
What about grouping aggregation, `CXformGbAgg2HashAgg::Exfp` and `CUtils::PdrgpcrGroupingKey` where we use FHashable since agg operator may request redistribution?

File: data/dxl/minidump/DQA-NonRedistributableCol.mdp (hunk 1 above)
Comment (khannaekta) [reply data/dxl/minidump/DQA-NonRedistributableCol.mdp]: You can run `psql -f /usr/local/gpdb/share/postgresql/contrib/hstore.sql` on GPDB master

**Comment by khannaekta** [comment]
@bhuvnesh2703 The problem this PR tries to solve is not to add a redistribute motion on a column that cannot used for distribution. GPDB passes 2 attribute value for each Type  `IsHashable` and `IsRedistributable` captured in the dxl metadata. Previously ORCA was using the `IsHashable` attribute of the type to compute for redistribution spec as well.
So for the function `CXformGbAgg2HashAgg::Exfp`and `CUtils::PdrgpcrGroupingKey`, we only wants to check if the column is hashable and not redistributable (as it is not concerned with the distribution spec at this point) and thus these functions do not need to be updated.

**Pull request merged**